src/Node/Internal.hs

```diff
@@ -603,6 +603,8 @@ startNode packingType peerData mkNodeEndPoint prng nodeEnv handlerIn handlerOut
                                 }
                       ; dispatcherThread <- async $
                             nodeDispatcher node handlerIn handlerOut
+                      -- Exceptions in the dispatcher are re-thrown here.
+                      ; link dispatcherThread
```

Comment:
I would have preferred to use a `withAsync` here but as far as I can tell it's not feasible due to the mutual dependence of `dispatcherThread` and `node`.